### PR TITLE
Parameters Converter

### PR DESCRIPTION
**Pull request by MAfarrag**
# Parameter Quantity Converter
- Add old parameter quantities in the `extold.ExtOldParametersQuantity`
- Add converter for the parameter quantities in the main_converter to the 

# New Features
- The main_converter.py now can convert the boundary condition quantities.

# Design
- Functions/method better have one return statement at the end of the function for readability (`ext.models.Boundary.forcing`).
- Test directory better be a mirror of the modules in the package directory.
- merge `base_converter.py`, `converter_factory.py` in `converters.py`.
- merge `tools.ext_old_to_new.enum_converters.py` to `tools.ext_old_to_new.utils`.
- the `__contains__` function is now a method `contains` in the `ConverterFactory` class.

# Dev
- Created `docs`, `pre-commit`,  group dependencies in the pyproject.toml

# Formating and Styling
None

# Testing

## Data
None

## New Tests
- `tests/dflowfm/test_extold.py::test_ext_old_parameter_quantity`
- `tests\tools\test_converters.py::TestConvertParameters::test_sample_data_file`
- `tests\tools\test_converters.py::TestConvertMeteo::test_default`

# Old errors
- Multiple uses of mutable objects like lists [] as a default values was converted to Field(default_factory=list)

### CODE REVIEW

File: hydrolib/tools/ext_old_to_new/converters.py [hunk 1]
```diff
@@ -118,6 +160,39 @@ def convert(self, forcing: ExtOldForcing) -> Boundary:
         return new_block
 
 
+def create_convert_inputs(forcing: ExtOldForcing) -> Dict[str, str]:
```
Comment (arthurvd) [inline hydrolib/tools/ext_old_to_new/converters.py:163]: Make the function name more descriptive + add docstring (I can reverse engineer that this one is probably shared functionality for initial and parameter fields (i.e., it is intended to produce a dict that can be used to construct an instance of some AbstractSpatialField subclass.)

File: hydrolib/tools/ext_old_to_new/converters.py [hunk 2]
```diff
@@ -151,35 +226,87 @@ def convert(self, forcing: ExtOldForcing) -> InitialField:
...
+        return new_block
+
+
+class ParametersConverter(BaseConverter):
+
+    def __init__(self):
+        super().__init__()
+
+    def convert(self, forcing: ExtOldForcing) -> ParameterField:
+        """Convert an old external forcing block with meteo data to a boundary
```
Comment (arthurvd) [inline hydrolib/tools/ext_old_to_new/converters.py:241]: docstring is (again ;-)) copy-past error from meteo.

File: hydrolib/tools/ext_old_to_new/converters.py [hunk 3]
```diff
@@ -151,35 +226,87 @@ def convert(self, forcing: ExtOldForcing) -> InitialField:
...
 
-        new_block = InitialField(**block_data)
+        Args:
+            forcing (ExtOldForcing): The contents of a single forcing block
+            in an old external forcings file. This object contains all the
+            necessary information, such as quantity, values, and timestamps,
+            required for the conversion process.
+
+        Returns:
+            Boundary: A Boindary object that represents the converted forcing
```
Comment (arthurvd) [inline hydrolib/tools/ext_old_to_new/converters.py:257]: typo

File: tests/tools/test_converters.py [hunk 4]
```diff
@@ -41,6 +43,41 @@ def test_polygon_data_file(self):
...
+            operand="O",
+        )
+
+        new_quantity_block = MeteoConverter().convert(forcing)
+        assert isinstance(new_quantity_block, Meteo)
+        assert new_quantity_block.quantity == "windx"
+        assert new_quantity_block.operand == "O"
+        assert new_quantity_block.forcingfile == DiskOnlyFileModel("windtest.amu")
+        assert new_quantity_block.forcingfiletype == "meteoGridEqui"
+        assert new_quantity_block.interpolationmethod == "linearSpaceTime"
```
Comment (arthurvd) [inline tests/tools/test_converters.py:78]: Would you not prefer:
```suggestion
        assert new_quantity_block.interpolationmethod == MeteoInterpolationMethod.linearSpaceTime        
```
and similar for all other enum values in the above assertions?

**Review by arthurvd — CHANGES_REQUESTED** [review]
Functionally it looks good. See my individual comments for some code considerations.

File: .github/pull_request_template.md
[the remark this answers is not in the document]
Comment (MAfarrag) [reply .github/pull_request_template.md:1]: - When merged to the main branch, it will be used automatically when you create a merge request
- Should i remove it from here?

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
+# Description
+
+- Please include a summary of the change and which issue is fixed.
+- Please also include relevant motivation and context.
+- List any dependencies that are required for this change.
+
+
+- Fixes # (issue)
+## Type of change
```
[the remark this answers is not in the document]
Comment (MAfarrag) [reply .github/pull_request_template.md:9]: just in case we needed to go back to any merge request and reopen it for any reason, all the needed information should be stored here in the merge

File: .github/pull_request_template.md
[the remark this answers is not in the document]
Comment (arthurvd) [reply .github/pull_request_template.md:1]: No need to remove now, but please take that approach in all future work :-)
regarding template and main: all clear!

**Review by arthurvd — APPROVED** [review]
I made some last fixes myself, all good now.
